pkg/modules/authorization.go

```diff
@@ -811,6 +814,49 @@ func InstallWithCerts(ctx context.Context, isDeleting bool, op utils.OperatorCon
 	return nil
 }
 
+// getAuthCrdDeploy - apply and deploy authorization crd manifest
+func getAuthCrdDeploy(op utils.OperatorConfig, cr csmv1.ContainerStorageModule) (string, error) {
+	yamlString := ""
+
+	auth, err := getAuthorizationModule(cr)
+	if err != nil {
+		return yamlString, err
+	}
+
+	authCrdPath := fmt.Sprintf("%s/moduleconfig/authorization/%s/%s", op.ConfigDirectory, auth.ConfigVersion, AuthCrds)
+	buf, err := os.ReadFile(filepath.Clean(authCrdPath))
+	if err != nil {
+		return yamlString, err
+	}
+
+	yamlString = string(buf)
+
+	yamlString = strings.ReplaceAll(yamlString, AuthNamespace, cr.Namespace)
+
+	return yamlString, nil
+}
+
+// AuthCrdDeploy - apply and delete Auth crds deployment
+func AuthCrdDeploy(ctx context.Context, op utils.OperatorConfig, cr csmv1.ContainerStorageModule, ctrlClient crclient.Client) error {
+	yamlString, err := getAuthCrdDeploy(op, cr)
+	if err != nil {
+		return err
+	}
+
+	ctrlObjects, err := utils.GetModuleComponentObj([]byte(yamlString))
```

Comment:
Instead of 
```
ctrlObjects, err := utils.GetModuleComponentObj([]byte(yamlString))
	if err != nil {
		return err
	}

	for _, ctrlObj := range ctrlObjects {
		if err := utils.ApplyObject(ctx, ctrlObj, ctrlClient); err != nil {
			return err
		}
	}
```

we can reuse the function `applyDeleteObjects`. This will also take care of deleting the CRDs

Reply:
Updated. Not sure why this function is inside of app mobility though. Probably it's better to move it in utils

Reply:
Yes I agree